tests/extrusion/test_real_tensorproduct.py

```diff
@@ -73,7 +73,7 @@ def quadrilateral(request):
 
 def test_vertical_average_variable(quadrilateral):
     """Test computing vertical average on mesh with variable nb of levels"""
-    tolerance = 1e-14
+    tolerance = 2e-14
```

Comment:
These tolerances are already tiny so I guess this isn't an issue. I am surprised that this is required for this PR to land though.

Reply:
I've changed this back. Let's see if the CI run passes now...

Reply:
Seems this change was necessary: https://github.com/firedrakeproject/firedrake/actions/runs/6876364089/job/18701826166?pr=3166

Reply:
OK. I think this is fine then.